packages/core/bootstrap/test/unit/cache.test.ts

```diff
@@ -136,5 +137,144 @@ describe('cache', () => {
       clock.tick(1000 * 30 + 1) // extra 1ms
       await callAndExpect(counter, 1, 1)
     })
+
+    it(`handles failed cache read, bypasses local and executes`, async () => {
+      jest.spyOn(AdapterCache.prototype, 'getResultForRequest').mockImplementation(() => {
+        throw 'asd'
+      })
+      const request: AdapterRequest = {
+        id: '1',
+        data: {},
+      }
+      const response = {
+        jobRunID: '1',
+        data: { jobRunID: '1', statusCode: 200, data: {}, result: 123 },
+        result: 123,
+        statusCode: 200,
+      }
+      const execute = async () => response
+      const middleware = await withCache()(execute, context)
+      const result = await middleware(request, {})
+      expect(result).toBe(response)
```

Comment:
Where does the `123` in the result come from? Are you sure the `expect(...).toBe(...)` check is also checking nested values?

Reply:
The execute function returns the response object, and in this case the expectation is that the middleware floats that up to the surface; the `toBe` function will compare the objects by reference so it's guaranteed they're the same instance